### PR TITLE
Feat/android/mute

### LINKED ISSUE #143: Optimize WindowInset Padding
On Android we want to optimize our content padding to do the following:

1. Allow the NavigationMapView (and underlying MapView) to fill the entire screen even under the device status bar/cameras etc.
2. Pad all control and information views with the device content padding.

It should be the default behavior to assume developers will want the map to extend under system controls instead of showing black bars around them. We should however, allow developers to use the contained behavior if desired (e.g. for a navigation view that is contained within a Navbar stack).

### LINKED ISSUE #151: Use string resources contentDescription in composables that use Icon
This will enable localization (separate, ongoing tasks for sure).

### LINKED ISSUE #159: Android - Mute button
UI component to mute / unmute spoken instructions.

### LINKED ISSUE #165: Android - Create default MapViewCamera for other navigation modes
Currently the `navigating()` MapViewCamera extension is tailored for automotive navigation. We should upgrade this methodology to allow specifying the a default navigation camera for the current activity (walking, cycling, driving).

Duplicate this ticket if needed (e.g. if iOS only is completed).

### PR DESCRIPTION
**Pull request by Archdoog**
- Adds mute button and mute functionality.
- Adds SystemUI disposable effect to hide the system bars (like other maps apps)
- Adds visual config to allow jetpack compose-like setup of controls through stack.
- Corrects faulty navigation camera view (we were missing padding on the companion style method).
- Refactors NavigationViews to match iOS (where a singular NavigationMapView is used for dynamic w/ overlays).

Closes #159 
Closes #165 
Closes #151 
Closes #143

### CODE REVIEW

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/runtime/HideSystemUIDisposableEffect.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+fun windowInsetsController(window: Window): WindowInsetsControllerCompat {
+  return WindowCompat.getInsetsController(window, window.decorView)
+}
+
+/**
+ * A Composable effect that hides the system UI (status bar and navigation bar) when the Composable
+ * is first composed and restores the system UI when the Composable is disposed.
+ */
+@Composable
+fun SystemUIDisposableEffect() {
```
Comment (ianthetechie) [inline android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/runtime/HideSystemUIDisposableEffect.kt:33]: Clever ;) I think the file name has a bit more descriptive type name though. Did you mean to name it `HideSystemUIDisposableEffect`?

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/runtime/NavigationCamera.kt [hunk 2]
```diff
@@ -6,22 +6,34 @@ import androidx.compose.ui.platform.LocalConfiguration
...
+
+  /** The recommended camera configuration for pedestrian navigation. */
+  data object Pedestrian : NavigationActivity(zoom = 20.0, pitch = 10.0)
+}
+
 /**
  * The camera configuration for navigation. This configuration sets the camera to track the user,
- * with a bearing of 18 degrees and a pitch of 45 degrees. It automatically adjusts the padding
- * based on the screen size and orientation.
+ * with a zoom of 16.0 and a pitch of 45 degrees. It automatically adjusts the padding based on the
```
Comment (ianthetechie) [inline android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/runtime/NavigationCamera.kt:22]: Maybe something like this so we don't have to update docstrings all the time ;0 I think the exact numbers probably _shouldn't_ matter to users of this composable.

```suggestion
 * with a high zoom level and moderate pitch for a 2.5D isometric view. It automatically adjusts the padding based on the
```

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/DynamicallyOrientingNavigationView.kt [hunk 3]
```diff
@@ -22,49 +34,68 @@ import com.stadiamaps.ferrostar.maplibreui.runtime.navigationMapViewCamera
...
+        locationRequestProperties,
+        onMapReadyCallback = { camera.value = navigationCamera },
+        content)
+
+    when (orientation) {
+      Configuration.ORIENTATION_LANDSCAPE -> {
+        LandscapeNavigationOverlayView(
+            modifier =
+                Modifier.fillMaxSize()
+                    // TODO: DO NOT MERGE - Finalize w/ device
```
Comment (ianthetechie) [inline android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/DynamicallyOrientingNavigationView.kt:79]: Noted 😂 These sorts of things always risk breakage...

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/DynamicallyOrientingNavigationView.kt [hunk 4]
```diff
@@ -22,49 +34,68 @@ import com.stadiamaps.ferrostar.maplibreui.runtime.navigationMapViewCamera
...
+            onTapExit = onTapExit)
+      }
+      else -> {
+        PortraitNavigationOverlayView(
+            modifier =
+                Modifier.fillMaxSize()
+                    .statusBarsPadding()
+                    .navigationBarsPadding()
+                    .padding(horizontal = 16.dp)
+                    .padding(bottom = 24.dp),
```
Comment (ianthetechie) [inline android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/DynamicallyOrientingNavigationView.kt:93]: Assume this is similar to the above. Seems like refactoring these as constants are the least we should do just so it's obvious what's going on. Might turn into something more elaborate like an interface later.

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/overlays/LandscapeNavigationOverlayView.kt [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import com.stadiamaps.ferrostar.core.NavigationViewModel
+import com.stadiamaps.ferrostar.core.mock.MockNavigationViewModel
+import com.stadiamaps.ferrostar.core.mock.pedestrianExample
+import com.stadiamaps.ferrostar.maplibreui.config.VisualNavigationViewConfig
+import com.stadiamaps.ferrostar.maplibreui.runtime.navigationMapViewCamera
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+@Composable
+fun LandscapeNavigationOverlayView(
```
Comment (ianthetechie) [inline android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/overlays/LandscapeNavigationOverlayView.kt:34]: Nice refactors 👍

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/config/MapControls.kt [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  return MapControls(
+      attribution =
+          AttributionSettings(
+              gravity = Gravity.BOTTOM or Gravity.END,
+              margins = MarginInsets(end = 270, bottom = bottom)),
+      compass = CompassSettings(enabled = false),
+      logo =
+          LogoSettings(
+              gravity = Gravity.BOTTOM or Gravity.END,
+              margins = MarginInsets(end = 32, bottom = bottom)))
```
Comment (ianthetechie) [inline android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/config/MapControls.kt:33]: How prone to breakage do you think these constants are? 😅

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/DynamicallyOrientingNavigationView.kt (hunk 3 above)
Comment (Archdoog) [reply android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/DynamicallyOrientingNavigationView.kt]: Exactly why I put this there. I just tested on a device and think it looks great 👍

**Review by ianthetechie — APPROVED** [review]

**Pull request merged**